docs/overview/architecture.md

```diff
@@ -113,4 +113,28 @@ When the user searches a vector from Vald:
 
 <!-- ### Update -->
 
-<!-- ### Delete -->
+<!-- ### Upsert -->
+
+### Delete
+
+<img src="../../assets/docs/delete_flow.png" />
+
+When the user delete a vector which is indexed in Vald Agent:
+
+1. Vald Ingress receives the delete request from the user. The request includes the vector ID(s) which is specified by the user.
+2. Vald Ingress will forward the request to the Vald Filter Gateway.
+3. Vald Filter Gateway will forward the request to the Vald Meta Gateway. Vald Meta Gateway is used to resolve the internal used UUID from Vald Meta to the user inserted vector ID in the Insert Step.
+4. Vald Meta Gateway will forward the request to the Vald Meta to confirm whether metadata which contains request ID(s) is existing or not.
+5. Vald Meta get the UUID(s) from the request ID(s). When the UUID(s) be not found, return with the error.
+6. If Vald Meta Gateway gets the UUID(s), Vald Meta Gateway will forward to the request with the UUID(s) to the Vald LB Gateway through the Vald Backup Gateway.
+7. Vald LB Gateway will broadcast the request to the Vald Agent. All pods of Vald Agent check the vector whose UUID is requested one is indexed or not. When idexing data is existing, Vald Agent will delete the vector and the UUID in an onmemory graph index.
+8. If Vald Agent successfully deletes the request data, it will return success to the Vald LB Gateway.
```

Comment:
```suggestion
8. If Vald Agent successfully deletes the request data, it will return ip addresses and name of Vald Agent to the Vald LB Gateway.
```